packages/react-strict-dom/src/native/modules/useStrictDOMElement.js

```diff
@@ -38,68 +40,89 @@ function proxy(nativeRef: React.RefObject<Node>, name: string) {
 export function useStrictDOMElement<T>(
   ref: React.RefSetter<Node>,
   { tagName }: Options
-): React.RefObject<T | null> {
-  const nativeRef = React.useRef<T | null>(null);
+): CallbackRef<T> {
   const { scale: viewportScale } = useViewportScale();
 
-  React.useImperativeHandle(ref, () => {
-    return {
-      get __nativeTag() {
-        const node = nativeRef.current as Node;
-        return node?.__nativeTag;
-      },
-      addEventListener: proxy(nativeRef, 'addEventListener'),
-      animate: proxy(nativeRef, 'animate'),
-      blur: proxy(nativeRef, 'blur'),
-      click: proxy(nativeRef, 'click'),
-      get complete() {
-        if (tagName === 'img') {
-          const node = nativeRef.current as Node;
-          if (node?.complete == null) {
-            // Assume images are never pre-loaded in React Native
-            return false;
+  const elementCallback = useElementCallback(
+    React.useCallback(
```

Comment:
I don't think you need `useElementCallback` here.

The `useCallback` already has `ref` as a dependency, which means the function returned by `useCallback` will already be invalidated if `ref` changes. And when that new function is passed into the native component's `ref` prop below, it will naturally detach the previous one and attach the new one.